plan/plan_loader_test.go

```diff
@@ -65,3 +66,65 @@ func TestLoadOneFile(t *testing.T) {
 	}
 	assert.Equal(t, gotPlans, expectPlans)
 }
+
+func TestPlan_ShouldReturnDeepCopyOfPlan(t *testing.T) {
+	pl := NewLoader(
+		func(blip.ConfigPlans) ([]blip.Plan, error) { return []blip.Plan{}, nil },
+	)
+	planName := "foobar"
+	pl.sharedPlans = []planMeta{
+		{
+			name: planName,
+			plan: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "1s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	got, err := pl.Plan("", planName, nil)
+	require.NoError(t, err)
+
+	expected := pl.sharedPlans[0].plan
+	assert.Equal(t, expected, got)
+	verifyDeepCopy(t, expected, got)
+}
+
+// verifyDeepCopy verifies is a is a deep copy of b by comparing address of the slices and maps in blip.Plan.
```

Comment:
Do we have to go that far: comparing addresses? Testing like this should both reproduce and verify the actual (right/wrong) output that motivated it: dupes. So we don't need to look at addresses; instead, we do what Blip did (get plan, sort, get plan again, sort again) then check: do we see this,
```
CommitLatency
CommitLatency
```
again (bug) or not (fixed)?

Reply:
Yeah I know 😞 . Unfortunately I could figure out a good way to test. The best ways I saw as verifying this issue was to either compare addresses (go does not give you a way to directly compare references) or to observe via side effects such as routines that would modify the `map` and `slice` fields of the structure. I don't feel strongly about having to compare addresses, so I'm more than open to verification that relies on attempting to modify the received plan and see if those changes are reflected original plan. However, I do strongly that we should not couple this with the source bug in the test for this package for the reasons noted in the comments above.

Since the topic of the verifying the exact bug in which this issue was surfaced, I will add a test for `ChangePlan` to capture this behavior.

Thoughts?